engine/resources/hetzner.go

```diff
@@ -0,0 +1,716 @@
+// Mgmt
+// Copyright (C) 2013-2021+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package resources
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/hetznercloud/hcloud-go/hcloud"
+	"github.com/purpleidea/mgmt/engine"
+	"github.com/purpleidea/mgmt/engine/traits"
+	"github.com/purpleidea/mgmt/util/errwrap"
+)
+
+const (
+	// HetznerPollLimit sets a lower limit on polling interval in seconds.
+	// Note that the hetzner API supports bursts at up to 3600 requests per hour.
+	// A higher polling interval is recommended when expecting high rates
+	// of operational http requests to the hetzner api.
+	HetznerPollLimit = 1
+	// WaitIntervalLimit sets a lower limit on wait intervals in seconds.
+	// Nonzero values disable burst requests to manage rate limit errors.
+	// A wait interval >= the polling interval is recommended.
+	WaitIntervalLimit = 0
```

Comment:
These constants are in the same namespace as the other resources so adding a Hetzner prefix might be a good idea...